packages/rum/src/domain/record/privacy.ts

```diff
@@ -1,91 +1,386 @@
- * sure we respect the privacy mode priorities.
+ * PUBLIC: Resolves the internal privacy level and remaps to level to the format
+ * exposed to the general codebase  (allow/mask/hidden/ignore) because
+ * NOT_SET/UNKNOWN/*-SEALED/MASK_FORMS_ONLY are not dev friendly and need not be handled by devs
  */
-export function getNodeInputPrivacyMode(node: Node, ancestorInputPrivacyMode: InputPrivacyMode): InputPrivacyMode {
-  if (!isElement(node)) {
-    return InputPrivacyMode.NONE
-  }
+export function getNodePrivacyLevel(node: Node, parentNodePrivacyLevel?: NodePrivacyLevelInternal): NodePrivacyLevel {
+  const privacyLevel = getInternalNodePrivacyLevel(node, parentNodePrivacyLevel)
+  return remapInternalPrivacyLevels(node, privacyLevel)
+}
 
-  const attribute = node.getAttribute(PRIVACY_ATTR_NAME)
+/**
+ * Remap the internal privacy levels to general use privacy levels
+ */
+export function remapInternalPrivacyLevels(
+  node: Node,
+  nodePrivacyLevelInternal: NodePrivacyLevelInternal
+): NodePrivacyLevel {
   if (
-    ancestorInputPrivacyMode === InputPrivacyMode.IGNORED ||
-    attribute === PRIVACY_ATTR_VALUE_INPUT_IGNORED ||
-    node.classList.contains(PRIVACY_CLASS_INPUT_IGNORED) ||
-    (isInputElement(node) && PRIVACY_INPUT_TYPES_TO_IGNORE.includes(node.type))
+    nodePrivacyLevelInternal === NodePrivacyLevelInternal.UNKNOWN ||
+    nodePrivacyLevelInternal === NodePrivacyLevelInternal.NOT_SET
   ) {
-    return InputPrivacyMode.IGNORED
+    // Fallback value from getInitialPrivacyLevel() needs to be remapped too for the `MASK_FORMS_ONLY` case.
+    nodePrivacyLevelInternal = getInitialPrivacyLevel()
   }
 
-  if (
-    ancestorInputPrivacyMode === InputPrivacyMode.MASKED ||
-    attribute === PRIVACY_ATTR_VALUE_INPUT_MASKED ||
-    node.classList.contains(PRIVACY_CLASS_INPUT_MASKED)
-  ) {
-    return InputPrivacyMode.MASKED
+  switch (nodePrivacyLevelInternal) {
+    case NodePrivacyLevelInternal.ALLOW:
+    case NodePrivacyLevelInternal.MASK:
+    case NodePrivacyLevelInternal.HIDDEN:
+    case NodePrivacyLevelInternal.IGNORE:
+      return nodePrivacyLevelInternal
+    // Conditional levels
+    case NodePrivacyLevelInternal.MASK_FORMS_ONLY:
+      return isFormElement(node) ? NodePrivacyLevel.MASK : NodePrivacyLevel.ALLOW
+  }
+  return NodePrivacyLevelInternal.HIDDEN
+}
+
+/**
+ * INTERNAL FUNC: Get privacy level without remapping (or setting cache)
+ * This function may be explicitly used when passing internal privacy levels to
+ * child nodes for performance reasons, otherwise you should use `getNodePrivacyLevel`
+ */
+export function getInternalNodePrivacyLevel(
+  node: Node,
+  parentNodePrivacyLevel?: NodePrivacyLevelInternal
+): NodePrivacyLevelInternal {
+  const isElementNode = isElement(node)
+
+  // Only Elements have tags directly applied
+  if (!isElementNode && node.parentElement) {
+    return parentNodePrivacyLevel || getInternalNodePrivacyLevel(node.parentElement)
+  }
+
```

Comment:
To me, it doesn't make sense to move the above referenced lines anywhere else.
The purpose here is two fold:
1. We expose a "fast path" to do less work for the common use case- text nodes.
2. The overall logic with of `getInternalNodePrivacyLevel` is to use `getNodeSelfPrivacyLevel` and `derivePrivacyLevelGivenParent`. They both need to operate on the same node.  I find (recursing?) up one level a clean strategy. Although `getNodeSelfPrivacyLevel` COULD handle text nodes, it doesn't because we don't pass text nodes in because we pass the node's parent- I think it's more pure of `getNodeSelfPrivacyLevel` to only consider it's own node. So in this sense getNodeSelfPrivacyLevel doesn't really handle non-element checks- even though it does handle other node types, like `DOCUMENT_NODE`- and could handle other node types like CDATA if we care.

(As a minor side note, yes we could limit the scope of `getNodeSelfPrivacyLevel` to only the Element type, but we don't need to so might as well keep it broad for future compatibility.)